### ISSUE TITLE
Flutter App is Crashes When using local engine build master branch.

### ISSUE THREAD

**Issue by canewsin** [flutter/flutter#83343]
## Steps to Reproduce

Followed all the steps here

https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment
https://github.com/flutter/flutter/wiki/Compiling-the-engine

when i compile host and for android-arm64 without any modifications, app is running fine.
but i modified java code and recompile it. App is crashing

log
```
2021-05-25 17:34:11.129 25363-25363/? A/flutter: [FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'
2021-05-25 17:34:11.228 10544-10544/? A/DEBUG: Abort message: '[FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'
    '
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #01 pc 00000000002b9414  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/lib/arm64/libflutter.so (BuildId: 8db0ac68ed04c246b699c26d2d3eaa6ae9c7b7c4)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #02 pc 00000000002d96fc  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/lib/arm64/libflutter.so (BuildId: 8db0ac68ed04c246b699c26d2d3eaa6ae9c7b7c4)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #03 pc 00000000002fddfc  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/lib/arm64/libflutter.so (BuildId: 8db0ac68ed04c246b699c26d2d3eaa6ae9c7b7c4)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #04 pc 0000000000305924  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/lib/arm64/libflutter.so (BuildId: 8db0ac68ed04c246b699c26d2d3eaa6ae9c7b7c4)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #05 pc 00000000002c508c  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/lib/arm64/libflutter.so (BuildId: 8db0ac68ed04c246b699c26d2d3eaa6ae9c7b7c4)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #13 pc 0000000000beaf4c  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.engine.FlutterJNI.performNativeAttach)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #16 pc 0000000000beb040  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.engine.FlutterJNI.attachToNative+12)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #19 pc 0000000000beabd0  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.engine.FlutterEngine.attachToJni+20)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #22 pc 0000000000beaa8e  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.engine.FlutterEngine.<init>+422)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #25 pc 0000000000beab88  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.engine.FlutterEngine.<init>+24)
2021-05-25 17:34:11.504 10544-10544/? A/DEBUG:       #28 pc 0000000000be3424  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.setupFlutterEngine+212)
2021-05-25 17:34:11.505 10544-10544/? A/DEBUG:       #31 pc 0000000000be2cbe  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onAttach+14)
2021-05-25 17:34:11.505 10544-10544/? A/DEBUG:       #34 pc 0000000000be3c82  /data/app/~~TSJCb5nqFdM5bG_-RhiUYg==/in.canews.movieroar-5PT7b27_7z4pSeyiU4eswA==/oat/arm64/base.vdex (io.flutter.embedding.android.FlutterActivity.onCreate+26)
```

**Expected results:** should run normally.

**Actual results:** App is crashing

**Comment by canewsin** [comment]
@TahaTesser could you assign/ping someone for the issue.

**Comment by canewsin** [comment]
https://github.com/flutter/flutter/issues/79782
found this issue, i was on master branch latest commit.

**Comment by TahaTesser** [comment]
Hi @canewsin 
I just fresh  `engine`  setup for Android using arm64 and updated my master setup to match, I can compile and run the app without modifications and also runs fine with modifications.  Changed keyboard types in the fine for Android for testing, changes are shown in the app.

Ran the app using the below 

`flutter run --local-engine-src-path /Volumes/Extreme/engine/src  --local-engine=android_debug_unopt_arm64`



> 
> 2021-05-25 17:34:11.129 25363-25363/? A/flutter: [FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'
> 2021-05-25 17:34:11.228 10544-10544/? A/DEBUG: Abort message: '[FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'
> 


This just says you've mismatch, please make sure you matching snapshots for your setup

**Comment by canewsin** [comment]
@TahaTesser don't know how you get that working directly, but I had stumbled upon it from two days, I found that I had to update dart-sdk in third-party/dart folder manually to master channel which is previously on dev channel, still it is not enough I just push the apk directly for almost 10 times without uninstalling app on device, but when I uninstall app on device and reinstall it through adb it works last night, but only after switching dart to master channel manually. I would consider it as an issue still now, even though app works.

**Comment by canewsin** [comment]
> Hi @canewsin 
> I just fresh  `engine`  setup for Android using arm64 and updated my master set up to match, I can compile and run the app without modifications and also runs fine with modifications.  Changed keyboard types for text input on Android for testing in the engine source, changes are shown in the app.
> 
> Ran the app using the below 
> 
> `flutter run --local-engine-src-path /Volumes/Extreme/engine/src  --local-engine=android_debug_unopt_arm64`
> 
> 
> 
> > 
> > 2021-05-25 17:34:11.129 25363-25363/? A/flutter: [FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'
> > 2021-05-25 17:34:11.228 10544-10544/? A/DEBUG: Abort message: '[FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'
> > 
> 
> 
> This just says you've mismatch, please make sure you matching snapshots for your setup 

And you are using debug build which has no .so libs to install, but I tried release builds, I think that would make the difference on app .so lib update issue.

**Comment by TahaTesser** [comment]
Hi @canewsin 
I just tried  again with new clean setup on release mode, it is not supported, running into https://github.com/flutter/flutter/issues/55950 
Confirmed [here](https://github.com/flutter/flutter/issues/55950#issuecomment-621609385)  and in your duplicate [issue](https://github.com/flutter/flutter/issues/83412) 

Debug mode works fine with or without modification


> 2021-05-25 17:34:11.228 10544-10544/? A/DEBUG: Abort message: '[FATAL:flutter/runtime/dart_vm_initializer.cc(22)] Error while initializing the Dart VM: Wrong full snapshot version, expected 'fb41b05fc6de686f3a2cb518e3dc7dcf' found 'e4a09dbf2bb120fe4674e0576617a0dc'

Doesn't reproduce

**Comment by darshankawar** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
